lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -17,96 +17,81 @@ def handles?(id)
-          # success, else it returns false
-          opts = { datacite: datacite_metadata.datacite_xml, _status: IDENTIFIER_STATUS[:public] }
-          opts[:_target] = location_uri unless location_uri.blank?
-          api_session.modify_identifier(id, opts)
-        end
-
-        # Following method will make a request to the EZID server to update the target URL associated
-        # with the EZID DOI entry on the server.
-        # (See _target in http://ezid.cdlib.org/doc/apidoc.html#internal-metadata)
-        # To delete the target URL stored in the DOI server, use an empty string as the argument.
-        # Uses the modify identifier API call, returns true if metadata update was successful
-        # if not, returns false
-        def update_doi_target_url(doi, target_url)
-          return false if doi.nil?
-          return false if target_url.nil?
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          api_session.modify_identifier(doi, _target: target_url, _status: IDENTIFIER_STATUS[:public])
+        # @return [Hash] hash containing the publishing info ready to be JSONified
+        def as_datacite_properties(digital_object, target_url)
+          HyacinthMetadata.as_datacite_properties(digital_object, target_url)
         end
 
-        def tombstone_impl(id)
-          return false if id.nil?
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          api_session.modify_identifier(id, _status: IDENTIFIER_STATUS[:unavailable])
+        # Returns true if an identifier exists in the external management system
+        def exists?(doi)
+          resp = rest_api.get_dois(doi)
+          rest_api.parse_doi_from_api_response_body(resp.body).present?
         end
 
-        private
-
-          # setup EZID API info: credentials, url, etc.
-          def api_session
-            Thread.current[:datacite_ezid_api_session] ||= Datacite::EzidSession.new(DATACITE[:user], DATACITE[:password])
-          end
-
-          def http_error_message(response)
-            return "" if response.nil?
-            "HTTP status code: #{response.http_status_code}\n" \
-            "HTTP server message: #{response.http_server_message}\n" \
-            "response body: #{response.body}"
+        # @param id [String]
+        # @param digital_object [DigitalObject]
+        # @param location_uri [String]
+        # @param doi_state [Symbol] doi_state can be set to one of the following: :draft, :findable, :registered
+        # @return [Boolean] true if update was successful, false otherwise
+        def update_impl(doi, digital_object:, location_uri:, doi_state: nil, **_args)
+          datacite_data = Datacite::RestApi::V2::Data.new(@prefix)
+          datacite_data.update_properties(as_datacite_properties(digital_object, location_uri))
```

Comment:
```suggestion
          datacite_data.update_properties(
            HyacinthMetadata.new(digital_object).as_datacite_properties(location_uri)
          )
```

Same optional note here as above about the two levels of wrapper methods.

Also a note that it says `location_url` in the earlier case and `location_uri` in this one.  Would be good to use the same thing in both cases.

Reply:
Are you thinking of `target_url` vs `location_uri`? That is the difference between Datacite/DOI language (`target_url`) and Hyacinth/Adapter language (`location_uri`). This may be a place for some YARD comments.

Reply:
Right!  I meant to type `target_url` and `location_uri`.  If both contexts should only refer to a url, and not a uri, maybe it would be better to use "url" in both cases so that no one assumes that any uri is valid as a value?

FWIW, the parameters on Hyacinth adapters aren't uniform because they serve different purposes, so no pressure to force "uri" naming just because some other adapters do.